pkg/config/config_test.go

```diff
@@ -847,6 +847,23 @@ env=["foo=bar"]`
 		gomega.Expect(config.Containers.EnableLabeledUsers).To(gomega.BeTrue())
 	})
 
+	It("HomeDirTest", func() {
+		oldHOMEDIR, set := os.LookupEnv("HOME")
+		dir, err := os.MkdirTemp("", "configTest")
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		defer os.RemoveAll(dir)
+
+		os.Setenv("HOME", dir)
+		_, err = defaultConfig()
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		if set {
+			os.Setenv("HOME", oldHOMEDIR)
+		} else {
+			os.Unsetenv("HOME")
+		}
+	})
+
```

Comment:
This is _completely unrepresentative_ of https://github.com/containers/podman/issues/23818 .

Ordinarily, if `$HOME` is an empty _but writable_ directory, `GetConfigHome` transparently creates `~/.config`, and nothing fails.

The reported failure is that `.config` does not exist _and the user does not have the permissions to create it_. In that case `GetConfigHome` fails.

Reply:
Yup.  Will work to fix.